### PR TITLE
PR for Add `zowe zos-files list zfs` command #429

### PR DESCRIPTION
**Pull request by cycaoecw**
Re-create PR for #429

### CODE REVIEW

**Comment by cycaoecw** [comment]
@tucker01 Hi, I open another PR for it, would you have a look?

**Comment by tucker01** [comment]
Hi @cycaoecw, thanks! Was out for a bit, but I'll make sure to review this in the next day or so...

File: packages/zosfiles/src/cli/list/zfs/Zfs.handler.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        else
+        {
+            response = await List.zfs(session, {
+                path: null,
+                fsname: commandParameters.arguments.fsname,
+                maxLength: commandParameters.arguments.maxLength
+            });
+        }
+
+        if (commandParameters.arguments.attributes && response.apiResponse.items.length > 0) {
```
Comment (tucker01) [inline packages/zosfiles/src/cli/list/zfs/Zfs.handler.ts]: I know some other commands have an `--attributes` argument, but it doesn't look like this is present on your command definition? Can this code be removed?

File: packages/zosfiles/src/cli/list/zfs/Zfs.handler.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            commandParameters.response.console.log(TextUtils.prettyJson(response.apiResponse.items));
+        } else {
+            commandParameters.response.data.setObj(response);
```
Comment (tucker01) [inline packages/zosfiles/src/cli/list/zfs/Zfs.handler.ts]: Can we move this line right after the API calls? This way it'll always get set?

**Comment by cycaoecw** [comment]
Hi @tucker01 thanks for your suggest! I have do some update, but not sure why codecov/patch is not pass. For if you supply neither fsname or path what is the behavior? It will list all mount point.

File: packages/zosfiles/src/cli/list/zfs/Zfs.handler.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            response = await List.zfs(session, {
+                path: null,
+                fsname: commandParameters.arguments.fsname,
+                maxLength: commandParameters.arguments.maxLength
+            });
+        }
+
+        commandParameters.response.data.setObj(response);
+
+        if (response.apiResponse.items.length > 0) {
```
Comment (tucker01) [inline packages/zosfiles/src/cli/list/zfs/Zfs.handler.ts]: This should probably be removed also... just outputting the table (via `commandParameters.response.format.output`) is sufficient for the list commands.

**Comment by tucker01** [comment]
I must have overlooked this, but it looks like there are no unit tests for `Zfs.handler.ts`? Probably why the codecov failed.

**Comment by cycaoecw** [comment]
@tucker01 yes, you are right, seems it really missing. let me check why

**Comment by cycaoecw** [comment]
@tucker01 I added back the unit test, but seems it still can not pass the codecov

**Comment by tucker01** [comment]
It looks like the unit tests for `Zfs.handler.ts` are incorrect? It looks like the file committed as `Zfs.handler.unit.test.ts` is a copy of the contents from `List.unit.test.ts`. The codecov report is showing the `Zfs.handler.ts` file to only be 10% tested. 

If you write some unit tests for `Zfs.handler.ts` I think we'd be good to go... you can model your handler unit tests after https://github.com/zowe/zowe-cli/blob/master/packages/zosfiles/__tests__/cli/list/am/AllMembers.handler.unit.test.ts. 

Sorry for the confusion. Let me know if you have more questions.

**Comment by cycaoecw** [comment]
Hi @tucker01 sorry that I am not aware the unit test files had re-structured. For the ZfsHandler unit test, I backup List.unit.test before and just using the backup as Zfs.handler.unit.test.ts last night. Let me update it and let you know.

**Comment by tucker01** [comment]
Sounds good. Thank you!

**Comment by cycaoecw** [comment]
Hi @tucker01 , sorry that take some times on it. I think should be ok now.

**Comment by tucker01** [comment]
No problem at all, there is no rush to finish anything 😄 . I'll take a look today.

**Pull request merged**